components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/AccessTokenIssuer.java

```diff
@@ -761,7 +761,20 @@ private void notifyImpersonation(OAuthTokenReqMessageContext tokReqMsgCtx)
         impersonationNotificationRequestDTO.setImpersonator(impersonatorUserId);
         AuthenticatedUser impersonatedUser = tokReqMsgCtx.getAuthorizedUser();
         impersonationNotificationRequestDTO.setSubject(impersonatedUser);
-        impersonationNotificationRequestDTO.setTenantDomain(tokReqMsgCtx.getAuthorizedUser().getTenantDomain());
+        String tenantDomain = impersonatedUser.getTenantDomain();
+        if (impersonatedUser.isFederatedUser() && impersonatedUser.getAccessingOrganization() != null) {
```

Comment:
Please add a comment saying, resolve the user tenant for a sub organization user in B2B SaaS app accessing case